Mage.Sets/src/mage/cards/l/LongRiverLurker.java

```diff
@@ -0,0 +1,140 @@
+
+        // Other Frogs you control have ward {1}.
+        this.addAbility(new SimpleStaticAbility(new GainAbilityControlledEffect(
+                new WardAbility(new GenericManaCost(1)), Duration.WhileOnBattlefield, filter, true)));
+
+        // When Long River Lurker enters, target creature you control can't be blocked this turn. Whenever that creature deals combat damage this turn, you may exile it. If you do, return it to the battlefield under its owner's control.
+        Ability ability = new EntersBattlefieldTriggeredAbility(new CantBeBlockedTargetEffect());
+        ability.addEffect(new LongRiverLurkerEffect());
+        ability.addTarget(new TargetControlledCreaturePermanent());
+        this.addAbility(ability);
+    }
+
+    private LongRiverLurker(final LongRiverLurker card) {
+        super(card);
+    }
+
+    @Override
+    public LongRiverLurker copy() {
+        return new LongRiverLurker(this);
+    }
+}
+
+class LongRiverLurkerEffect extends OneShotEffect {
+
+    LongRiverLurkerEffect() {
+        super(Outcome.Benefit);
+        staticText = "Whenever that creature deals combat damage this turn, you may exile it. If you do, return it to the battlefield under its owner's control.";
+    }
+
+    private LongRiverLurkerEffect(final LongRiverLurkerEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public LongRiverLurkerEffect copy() {
+        return new LongRiverLurkerEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = game.getPermanent(source.getFirstTarget());
+        if (permanent == null) {
+            return false;
+        }
+        game.addDelayedTriggeredAbility(new LongRiverLurkerTriggeredAbility(new MageObjectReference(permanent, game)), source);
+        return true;
+    }
+}
+
+class LongRiverLurkerTriggeredAbility extends DelayedTriggeredAbility {
+
+    private final MageObjectReference mor;
+
+    LongRiverLurkerTriggeredAbility(MageObjectReference mor) {
+        super(new ExileThenReturnTargetEffect(false, false).setTargetPointer(new FixedTarget(mor)),
+                Duration.EndOfTurn, false, true);
+        this.mor = mor;
+    }
+
+    private LongRiverLurkerTriggeredAbility(final LongRiverLurkerTriggeredAbility ability) {
+        super(ability);
+        this.mor = ability.mor;
+    }
+
+    @Override
+    public LongRiverLurkerTriggeredAbility copy() {
+        return new LongRiverLurkerTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DAMAGED_BATCH_FOR_ALL;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        int amount = ((DamagedBatchAllEvent) event)
+                .getEvents()
+                .stream()
+                .filter(DamagedEvent::isCombatDamage)
+                .filter(e -> mor.refersTo(e.getAttackerId(), game))
```

Comment:
wouldn't anyMatch() here be sufficient? I would think all events have amount >= 1

Reply:
Not sure, I just copied the `DealsCombatDamageTriggeredAbility` code for that. I guess since this effect doesn't use the amount of damage being dealt it's fine to use anyMatch?

Reply:
ah yeah that would be why